tools/format_python_tools.py

```diff
@@ -0,0 +1,59 @@
+import argparse
+import fnmatch
+import os
+import sys
+
+from yapf.yapflib.yapf_api import FormatFile
+
+
+def collect_files():
+  """Collect all Python files in the tools directory."""
+  matches = []
+  for root, dirnames, filenames in os.walk('.'):
+    for filename in fnmatch.filter(filenames, '*.py'):
+      matches.append(os.path.join(root, filename))
+  return matches
+
+
+def validate_format(fix=False):
+  """Check the format of python files in the tools directory.
+
+    Arguments:
+        fix: a flag to indicate if fixes should be applied.
+    """
+  fixes_required = False
+  failed_update_files = set()
+  successful_update_files = set()
+  for file in collect_files():
+    reformatted_source, encoding, changed = FormatFile(
+        file, style_config='.style.yapf', in_place=fix, print_diff=not fix)
+    if not fix:
+      fixes_required = changed
+      if reformatted_source:
+        print(reformatted_source)
```

Comment:
What do errors look like? It'd be nice if it issued them in compiler-like format, e.g.:

tools/foo.py: 3: error ... suggested fix ...

OTOH the usage model might be such that if there are any errors the only sane thing to do is to use the tool to fix them and not worry about what changes it is actually making. So I wouldn't hold up this PR to bike-shed on that.

Reply:
Errors look similar to what you've outlined.
```bash
Running Python format check...
--- ./check_format.py	(original)
+++ ./check_format.py	(reformatted)
@@ -21,7 +21,8 @@
 PROTO_SUFFIX = (".proto")
 
 # Files in these paths can make reference to protobuf stuff directly
-GOOGLE_PROTOBUF_WHITELIST = ("ci/prebuilt", "source/common/protobuf", "api/test")
+GOOGLE_PROTOBUF_WHITELIST = ("ci/prebuilt", "source/common/protobuf",
+                             "api/test")
 REPOSITORIES_BZL = "bazel/repositories.bzl"
 
 # Files matching these exact names can reference real-world time. These include the class
@@ -54,22 +55,34 @@
 
 PROTOBUF_TYPE_ERRORS = {
     # Well-known types should be referenced from the ProtobufWkt namespace.
-    "Protobuf::Any":                    "ProtobufWkt::Any",
-    "Protobuf::Empty":                  "ProtobufWkt::Empty",
-    "Protobuf::ListValue":              "ProtobufWkt::ListValue",
-    "Protobuf::NULL_VALUE":             "ProtobufWkt::NULL_VALUE",
-    "Protobuf::StringValue":            "ProtobufWkt::StringValue",
-    "Protobuf::Struct":                 "ProtobufWkt::Struct",
-    "Protobuf::Value":                  "ProtobufWkt::Value",
+    "Protobuf::Any":
+    "ProtobufWkt::Any",
+    "Protobuf::Empty":
+    "ProtobufWkt::Empty",
+    "Protobuf::ListValue":
+    "ProtobufWkt::ListValue",
+    "Protobuf::NULL_VALUE":
+    "ProtobufWkt::NULL_VALUE",
+    "Protobuf::StringValue":
+    "ProtobufWkt::StringValue",
+    "Protobuf::Struct":
+    "ProtobufWkt::Struct",
+    "Protobuf::Value":
+    "ProtobufWkt::Value",
 
     # Maps including strings should use the protobuf string types.
-    "Protobuf::MapPair<std::string":    "Protobuf::MapPair<Envoy::ProtobufTypes::String",
+    "Protobuf::MapPair<std::string":
+    "Protobuf::MapPair<Envoy::ProtobufTypes::String",
 
     # Other common mis-namespacing of protobuf types.
-    "ProtobufWkt::Map":                 "Protobuf::Map",
-    "ProtobufWkt::MapPair":             "Protobuf::MapPair",
-    "ProtobufUtil::MessageDifferencer": "Protobuf::util::MessageDifferencer"
+    "ProtobufWkt::Map":
+    "Protobuf::Map",
+    "ProtobufWkt::MapPair":
+    "Protobuf::MapPair",
+    "ProtobufUtil::MessageDifferencer":
+    "Protobuf::util::MessageDifferencer"
 }
+
 
 # lookPath searches for the given executable in all directories in PATH
 # environment variable. If it cannot be found, empty string is returned.
@@ -80,10 +93,12 @@
       return executable_path
   return ""
```